tests/host_tools/cargo_build.py

```diff
@@ -28,26 +28,41 @@
 CARGO_UNITTEST_REL_PATH = os.path.join(CARGO_BUILD_REL_PATH, "test")
 
 
+def cargo(
+    subcommand,
+    cargo_args: str = "",
+    subcommand_args: str = "",
+    *,
+    env: dict = None,
+    cwd: str = None,
+):
+    """Executes the specified cargo subcommand"""
+    env = env or {}
+
+    env_string = " ".join(f'{key}="{str(value)}"' for key, value in env.items())
```

Comment:
pity that run_cmd doesn't accept a `dict` for end. It could be passed straight to `subprocess.run`

Reply:
haha, one step at a time, lets leave that for a follow up xP